docs/user-guide/quickstarts/generic.md

```diff
@@ -0,0 +1,130 @@
+$ git clone git@github.com:<YOUR_USERNAME_HERE>/quickstart-generic.git
+$ cd quickstart-generic/
+```
+
+## Developing the app
+
+Now that we’ve setup our app we can start developing. This app demonstrates how to run a `Makefile` and bash script (`my_script.sh`) in your Screwdriver build.
+
+### screwdriver.yaml
+
+The `screwdriver.yaml` is the only config file you need for using Screwdriver. In it, you will define all your steps needed to successfully develop, build and deploy your application.
+
+#### Workflow
+
+The `workflow` describes the order that the jobs execute. The "main" job, which is created by default, is always
+executed first, followed by jobs listed in this workflow block.
+
+Here, we have defined a job named "second_job", which
+will run after the "main" job.
+
+```yaml
+---
+# Workflow list definition
+workflow:
+  - second_job
+```
+
+#### Shared
+The `shared` section is where you would define any attributes that all your jobs will inherit.
+
+In our example, we state that all our jobs will run in the same Docker container image "buildpack-deps". The `image` is usually defined in the form of "repo_name". Alternatively, you can define the image as "repo_name:tag_label", where "tag_label" is a version. See the [Docker documentation](https://hub.docker.com/r/library/buildpack-deps/) for more information.
+
+```yaml
+# Shared definition block
+shared:
+  image: buildpack-deps
+```
+
+#### Jobs
+The `jobs` section is where all the tasks (or `steps`) that each job will execute is defined. All pipelines have "main" implicitly defined. The definitions in your screwdriver.yaml file will override the implied defaults.
+
+### Steps
+The `steps` section contains a list of commands to execute.
+Each step takes the form "step_name: command_to_run". The "step_name" is a convenient label to reference it by. The
+"command_to_run" is the single command that is executed during this step.
+
+In our example, our "main" job executes a simple piece of inline bash code. We also define another job called "second_job". In this job, we intend on running a different set of commands. The "make_target" step calls a Makefile target to perform some set of actions. This is incredibly useful when you need to perform a multi-line command.
+The "run_arbitrary_script" executes a script. This is an alternative to a Makefile target where you want to run a series of commands related to this step.
+
+```yaml
+# Job definition block
+# "main" is a default job that all pipelines have
+jobs:
+  main:
+    # Steps definition block.
+    steps:
+      - hello: echo hi world
+  second_job:
+    steps:
+      - make_target: make greetings
+      - run_arbitrary_script: ./my_script.sh
+```
+
+Now that we have a working repository, let's head over to the Screwdriver UI to build and deploy an app.
+
+## Building with Screwdriver
+
+In order to use Screwdriver, you will need to login to Screwdriver using Github, set up your pipeline, and start a build.
+
+### Login with Github
```

Comment:
This may not really need to be a separate step. The login flow will be forced upon them when clicking on create.